packages/graalvm-kt/src/main/kotlin/elide/runtime/plugins/kotlin/Extensions.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 Elide Ventures, LLC.
+ *
+ * Licensed under the MIT license (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   https://opensource.org/license/mit/
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under the License.
+ */
+
+package elide.runtime.plugins.kotlin
+
+import org.graalvm.polyglot.Source
+import java.io.File
+import elide.runtime.core.DelicateElideApi
+import elide.runtime.core.PolyglotContext
+import elide.runtime.core.PolyglotValue
+import elide.runtime.core.evaluate
+
+/**
+ * Execute the given Kotlin [source] code as an interactive snippet, returning the result. This is equivalent to
+ * calling [PolyglotContext.evaluate] and selecting [Kotlin] as source language with an interactive source.
+ *
+ * @param source The Kotlin snippet to be evaluated.
+ * @return The result of the invocation.
+ */
+@DelicateElideApi public fun PolyglotContext.kotlin(source: String, name: String? = null): PolyglotValue {
+  return evaluate(
+    Source.newBuilder(Kotlin.languageId, source, name ?: "snippet.kts")
+      .interactive(true)
+      .build(),
+  )
+}
```

Comment:
Synthetic languages can be used in the same way as first-class ones: it doesn't matter that the language id `"kt"` is not supported by GraalVM, since these sources will never reach the GraalVM context (as long as the proper plugins are installed).